### DISCUSSION TITLE
Creating UEFI Boot ISO Image

### DISCUSSION THREAD

**Discussion by glenc2004** [ipxe/ipxe#1270] [Q&A]
Hello. I need some assistance in recreating a Bootable UEFI ISO image that contains a built in menu to ask the user for Networking information and then connecting to a web server to serve up the rest of the options for installation of Redhat Linux System on a VMWare system. I unfortunately don't have access other than CDROM so I need an ISO image. I built this before and was successful is using code from the repository from about 2 years ago. Unfortunately my DEV environment was deleted and I lost all my configs for iPXE and also unfortunately (and stupid of me) not documenting my build process so it could have been recreated.

I have cloned a fresh copy of iPXE and made a few changes to general.h to turn on scripting and multiboot (I don't remember if that was needed or not) and the resulting ISO image does not contain the EFI folder/directory containing the EFI code to boot. I tried to boot it anyway and from the brief prompt I see the boot was unsuccessful and then goes back to the boot selection screen for EFI.

I have search and googled myself to death and have not found any searches where anyone was asking for this specific case. I know we really don't need to do very many changes to the configs to have a successful build. But this is eluding me as how to make a successful ISO image for booting EFI. This took me a lot of time before and now once again I'm trying to get back in touch with the iPXE build process as it's been a long while.

This is my latest definition I've played with.

`/*
 * Image types
 *
 * Etherboot supports various image formats.  Select whichever ones
 * you want to use.
 *
 */
//#define   IMAGE_NBI       /* NBI image support */
//#define   IMAGE_ELF       /* ELF image support */
//#define   IMAGE_MULTIBOOT     /* MultiBoot image support */
//#define   IMAGE_PXE       /* PXE image support */
#define IMAGE_SCRIPT        /* iPXE script image support */
//#define   IMAGE_BZIMAGE       /* Linux bzImage image support */
//#define   IMAGE_COMBOOT       /* SYSLINUX COMBOOT image support */
//#define   IMAGE_EFI       /* EFI image support */
//#define   IMAGE_SDI       /* SDI image support */
//#define   IMAGE_PNM       /* PNM image support */
#define IMAGE_PNG       /* PNG image support */
#define IMAGE_DER       /* DER image support */
#define IMAGE_PEM       /* PEM image support */
//#define   IMAGE_ZLIB      /* ZLIB image support */
//#define   IMAGE_GZIP      /* GZIP image support */
//#define   IMAGE_UCODE     /* Microcode update image support */
`
And this is my build commands:

`make EMBED=menu.ipxe bin/ipxe.lkrn bin-x86_64-efi/ipxe.efi`
`./util/genfsimg -o ipxe.iso bin/ipxe.lkrn bin-x86_64-efi/ipxe.efi`

Kicking myself for not keeping this config someplace safe but oh well.

If anyone can help it would be appreciated.

Thank you.

**Comment by NiKiZe** [comment]
Why are you modifying the config at all? Unless you have something that doesn't work, don't touch it.
